pkg/api/nexus.go

```diff
@@ -30,3 +32,53 @@ func NewRaftReplicator(store db.Store, opts ...raft.Option) (RaftReplicator, err
 		return internal_raft.NewReplicator(store, options), nil
 	}
 }
+
+const RecordSeparator = byte(0x1E)
+
+func (req *SaveRequest) Encode() ([]byte, error) {
+	var argBuf bytes.Buffer
+	if err := gob.NewEncoder(&argBuf).Encode(req.Args); err != nil {
+		return nil, err
+	}
+	res := bytes.NewBuffer(req.Data)
+	res.WriteByte(RecordSeparator)
+	_, _ = res.ReadFrom(&argBuf)
+	return res.Bytes(), nil
+}
+
+func (req *SaveRequest) Decode(data []byte) error {
+	sepIndex := bytes.IndexByte(data, RecordSeparator)
+	if sepIndex < 0 {
+		return errors.New("invalid SaveRequest to decode, no record separator found")
+	}
+
+	req.Data = data[0:sepIndex]
+	req.Args = make(map[string][]byte)
+	buf := bytes.NewBuffer(data[sepIndex+1:])
+	err := gob.NewDecoder(buf).Decode(&req.Args)
+	return err
```

Comment:
Wouldn't using msgpack/protobuf allow this to be binary compatible with future changes. Using a separator can cause it be incorrect in case the data contains that same separator byte inside data.

Reply:
Same as my previous reply. Also note that the types `SaveRequest` and `LoadRequest` are currently used only by the Nexus `repl` program. That is, only those implementations that depend on the Nexus GRPC APIs, which is usually rare. `dkv` directly binds with Nexus and does NOT use these types (and their encode/decode methods).